src/Features/CSharp/Portable/Diagnostics/Analyzers/CSharpUseImplicitTypeDiagnosticAnalyzer.cs

```diff
@@ -107,9 +107,18 @@ protected override bool TryAnalyzeVariableDeclaration(TypeSyntax typeName, Seman
                 }
 
                 var variable = variableDeclaration.Variables.Single();
-                if (AssignmentSupportsStylePreference(
-                        variable.Identifier, typeName, variable.Initializer.Value,
-                        semanticModel, optionSet, cancellationToken))
+                var initializer = variable.Initializer.Value;
+
+                // Discourage using "var pointer = stackalloc", after introducing C# 7.2, as it might be confusing to users expecting a Span<>
```

Comment:
❗️ This analyzer isn't the place to do this. The new C# 7.2 features do not affect the use of `var` from the perspective of this analyzer.

Reply:
@sharwell can you please explain? the IDE should never suggest that particular pattern in that case.